pytest failing on fastapi and pyspark tests on master

When I build a clean contributor environment as per Conda instructions https://github.com/pandera-dev/pandera/blob/main/.github/CONTRIBUTING.md, and run 'pytest tests' on the master branch, I get the following test failures. I didn't have this problem the last time I contributed in May/June timeframe. I also tried virtualenv to the same effect. Are there extra steps in environment setup that I'm missing?

```
======================================================================================================================================================================================= short test summary info =======================================================================================================================================================================================
FAILED tests/fastapi/test_app.py::test_items_endpoint - requests.exceptions.ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=8000): Max retries exceeded with url: /items/ (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x156868f10>: Failed to establish a new connection: [Errno 61] Connection refused'))
FAILED tests/fastapi/test_app.py::test_transactions_endpoint - requests.exceptions.ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=8000): Max retries exceeded with url: /transactions/ (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x1574ecee0>: Failed to establish a new connection: [Errno 61] Connection refused'))
FAILED tests/fastapi/test_app.py::test_upload_file_endpoint - requests.exceptions.ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=8000): Max retries exceeded with url: /file/ (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x157bc21c0>: Failed to establish a new connection: [Errno 61] Connection refused'))
FAILED tests/pyspark/test_pyspark_check.py::TestCustomCheck::test_extension - AttributeError: '<class 'pandera.api.checks.Check'>' object has no attribute 'new_pyspark_check'. Make sure any custom checks have been registered using the extensions api.
FAILED tests/pyspark/test_pyspark_check.py::TestCustomCheck::test_extension_pydantic - pandera.errors.SchemaInitError: custom check 'new_pyspark_check' is not available. Make sure you use pandera.extensions.register_check_method decorator to register your custom check method.
FAILED tests/pyspark/test_schemas_on_pyspark_pandas.py::test_dataframe_schema_case[True] - py4j.protocol.Py4JJavaError: An error occurred while calling o15883.collectToPython.
FAILED tests/pyspark/test_schemas_on_pyspark_pandas.py::test_dataframe_schema_case[False] - py4j.protocol.Py4JJavaError: An error occurred while calling o16622.collectToPython.
FAILED tests/pyspark/test_schemas_on_pyspark_pandas.py::test_index_dtypes[Index-True-str] - py4j.protocol.Py4JJavaError: An error occurred while calling o31793.collectToPython.
FAILED tests/pyspark/test_schemas_on_pyspark_pandas.py::test_index_dtypes[Index-False-str] - py4j.protocol.Py4JJavaError: An error occurred while calling o36641.collectToPython.
FAILED tests/pyspark/test_schemas_on_pyspark_pandas.py::test_index_dtypes[MultiIndex-True-str] - py4j.protocol.Py4JJavaError: An error occurred while calling o43865.collectToPython.
FAILED tests/pyspark/test_schemas_on_pyspark_pandas.py::test_index_dtypes[MultiIndex-False-str] - py4j.protocol.Py4JJavaError: An error occurred while calling o51089.collectToPython.
FAILED tests/pyspark/test_schemas_on_pyspark_pandas.py::test_dtype_coercion[str-str] - py4j.protocol.Py4JJavaError: An error occurred while calling o74410.collectToPython.
FAILED tests/pyspark/test_schemas_on_pyspark_pandas.py::test_schema_model - py4j.protocol.Py4JJavaError: An error occurred while calling o77287.collectToPython.
FAILED tests/pyspark/test_schemas_on_pyspark_pandas.py::test_check_comparison_operators[check9-a-b] - py4j.protocol.Py4JJavaError: An error occurred while calling o79944.collectToPython.
FAILED tests/pyspark/test_schemas_on_pyspark_pandas.py::test_check_comparison_operators[check10-faa-foo] - py4j.protocol.Py4JJavaError: An error occurred while calling o80032.collectToPython.
FAILED tests/pyspark/test_schemas_on_pyspark_pandas.py::test_check_comparison_operators[check11-ab-ba] - py4j.protocol.Py4JJavaError: An error occurred while calling o80120.collectToPython.
FAILED tests/pyspark/test_schemas_on_pyspark_pandas.py::test_check_comparison_operators[check12-ba-ab] - py4j.protocol.Py4JJavaError: An error occurred while calling o80208.collectToPython.
========================================================================================================================================================= 17 failed, 2809 passed, 84 skipped, 8 xfailed, 123818 warnings in 581.28s (0:09:41) =========================================================================================================================================================
```

Heya @cosmicBboy most issues are gone now (thanks!). The first two below show up every time, and the third one shows up every once in a while.

```
=============================================================================================================================== short test summary info ================================================================================================================================
FAILED tests/pyspark/test_pyspark_check.py::TestCustomCheck::test_extension - AttributeError: '<class 'pandera.api.checks.Check'>' object has no attribute 'new_pyspark_check'. Make sure any custom checks have been registered using the extensions api.
FAILED tests/pyspark/test_pyspark_check.py::TestCustomCheck::test_extension_pydantic - pandera.errors.SchemaInitError: custom check 'new_pyspark_check' is not available. Make sure you use pandera.extensions.register_check_method decorator to register your custom check method.
FAILED tests/pyspark/test_schemas_on_pyspark_pandas.py::test_nullable[dtype10] - IndexError: tuple index out of range
================================================================================================= 3 failed, 2859 passed, 83 skipped, 13 xfailed, 109157 warnings in 516.28s (0:08:36) ==================================================================================================
```

Details for the third error are as follows
```
        else:
            try:
                ks_null_sample: ps.DataFrame = ps.DataFrame(null_sample)
            except TypeError as exc:
                match = re.search(
>                   r"can not accept object (<NA>|NaT) in type", exc.args[0]
                )
E               IndexError: tuple index out of range
E               Falsifying example: test_nullable(
E                   dtype=DataType(timedelta64[ns]),
E                   data=data(...),
E               )
E               Draw 1: 
E                      field
E                   0    NaT
E                   1 0 days
E                   2 0 days
E                   3 0 days
E                   4 0 days
E               
E               Draw 2: 
E                      field
E                   0 0 days
E                   1 0 days
E                   2 0 days
E                   3 0 days
E                   4 0 days
E               
E               Explanation:
E                   These lines were always and only run by failing examples:
E                       /Users/derinw/miniforge3/envs/pandera-dev/lib/python3.10/site-packages/pandas/core/arrays/_mixins.py:398
E                       /Users/derinw/miniforge3/envs/pandera-dev/lib/python3.10/site-packages/pandas/core/dtypes/missing.py:714
E                       /Users/derinw/miniforge3/envs/pandera-dev/lib/python3.10/site-packages/pandas/core/internals/blocks.py:1456
E                       /Users/derinw/miniforge3/envs/pandera-dev/lib/python3.10/site-packages/pandas/core/internals/blocks.py:246
E                       /Users/derinw/miniforge3/envs/pandera-dev/lib/python3.10/site-packages/pandas/core/internals/blocks.py:247
E                       (and 4 more with settings.verbosity >= verbose)

tests/pyspark/test_schemas_on_pyspark_pandas.py:344: IndexError
```

--- a/tests/pyspark/test_pyspark_check.py
+++ b/tests/pyspark/test_pyspark_check.py
@@ -3,6 +3,7 @@
 # pylint:disable=abstract-method
 import datetime
 import decimal
+from unittest import mock
 
 from pyspark.sql.functions import col
 from pyspark.sql.types import (
@@ -32,6 +33,29 @@ from pandera.pyspark import DataFrameModel, Field
 from pandera.backends.pyspark.decorators import validate_scope
 from pandera.pyspark import DataFrameSchema, Column
 from pandera.errors import PysparkSchemaError
+
+
+@pytest.fixture(scope="function")
+def extra_registered_checks():
+    """temporarily registers custom checks onto the Check class"""
+
+    # pylint: disable=unused-variable
+    with mock.patch(
+        "pandera.Check.REGISTERED_CUSTOM_CHECKS", new_callable=dict
+    ):
+
+        @pandera.extensions.register_check_method
+        def new_pyspark_check(pyspark_obj, *, max_value) -> bool:
+            """Ensure values of a series are strictly below a maximum value.
+            :param data: PysparkDataframeColumnObject column object which is a contains dataframe and column name to do the check
+            :param max_value: Upper bound not to be exceeded. Must be
+                a type comparable to the dtype of the column datatype of pyspark
+            """
+            # test case exists but not detected by pytest so no cover added
+            cond = col(pyspark_obj.column_name) <= max_value
+            return pyspark_obj.dataframe.filter(~cond).limit(1).count() == 0
+
+        yield
 
 
 class TestDecorator:
@@ -1598,19 +1622,9 @@ class TestCustomCheck(BaseClass):
             if df_out.pandera.errors:
                 raise PysparkSchemaError
 
-    @staticmethod
-    @pandera.extensions.register_check_method
-    def new_pyspark_check(pyspark_obj, *, max_value) -> bool:
-        """Ensure values of a series are strictly below a maximum value.
-        :param data: PysparkDataframeColumnObject column object which is a contains dataframe and column name to do the check
-        :param max_value: Upper bound not to be exceeded. Must be
-            a type comparable to the dtype of the column datatype of pyspark
-        """
-        # test case exists but not detected by pytest so no cover added
-        cond = col(pyspark_obj.column_name) <= max_value
-        return pyspark_obj.dataframe.filter(~cond).limit(1).count() == 0
-
-    def test_extension(self, spark):
+    def test_extension(
+        self, spark, extra_registered_checks
+    ):  # pylint: disable=unused-argument
         """Test custom extension with DataFrameSchema way of defining schema"""
         schema = DataFrameSchema(
             {
@@ -1631,7 +1645,9 @@ class TestCustomCheck(BaseClass):
             IntegerType(),
         )
 
-    def test_extension_pydantic(self, spark):
+    def test_extension_pydantic(
+        self, spark, extra_registered_checks
+    ):  # pylint: disable=unused-argument
         """Test custom extension with DataFrameModel way of defining schema"""
 
         class Schema(DataFrameModel):

